server.go

```diff
@@ -319,6 +319,8 @@ func Serve(opts *ServeConfig) {
 			ClientAuth:   tls.RequireAndVerifyClientCert,
 			ClientCAs:    clientCertPool,
 			MinVersion:   tls.VersionTLS12,
+			RootCAs:      clientCertPool,
+			ServerName:   "localhost",
 		}
```

Comment:
Not related to this change, but I also noticed that the error messages on L307 and L313 might be wrong. They should be referring to "server certificate" and not "client certificate" I believe.

Reply:
Oh yeah, good catch. I noticed that too but forgot to change it.